numba/np/arraymath.py

```diff
@@ -4422,3 +4422,45 @@ def impl(a, b):
         return _cross2d_operation(a_, b_)
 
     return impl
+
+
+#----------------------------------------------------------------------------
+# Equality tests implemented on Floats
+# Implemented based on this feature request
+# - https://github.com/numba/numba/issues/5977
+@overload(np.isclose)
+def np_isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+    if not type_can_asarray(a) or not type_can_asarray(b):
+        raise TypingError("Inputs a and b must be array-like.")
+    if not isinstance(rtol, types.Float) or not isinstance(atol, types.Float):
+        raise TypingError("Relative and absolute tolerance must " +
+                          "be represented as floats.")
+    if not isinstance(equal_nan, types.Boolean):
+        raise TypingError("Equal_nan must be cast to a bool.")
+
+    def impl(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+        a_ = np.asarray(a)
+        b_ = np.asarray(b)
+        return _isclose(a_, b_, rtol, atol, equal_nan)
+
+    return impl
+
+
+@register_jitable
+def _isclose(a, b, rtol, atol, equal_nan):
+    abs_diff = np.absolute(a - b)
+    min_tolerance = (rtol * np.absolute(b) + atol)
+    y = abs_diff <= min_tolerance
+    # Opposite values of infinity will produce an equality in the
+    # prev line, so check for no infinities. We only need to check one side
+    # because the LHS will always be either inf or NaN if there is an inf
+    y = np.logical_and(y, np.logical_not(np.isinf(abs_diff)))
+    # Python default behavior is to have NaN comparison be False,
+    # so handle equal_nan separately
+    if equal_nan:
+        # Logically and together NaN between A and B and Or it with y
+        nans = np.logical_and(np.isnan(a), np.isnan(b))
+        y = np.logical_or(y, nans)
+    # Add an extra equality check in case there is an inf
+    equality = a == b
+    return np.logical_or(y, equality)
```

Comment:
Be careful to match NumPy's logic, even down to order of operations. e.g. NumPy does `atol + rtol * abs(b)`.
It also shortcuts for the all-finite comparison case. For reference: https://github.com/numpy/numpy/blob/97d2db483fc0ffd46f38d0e1c39d5fc001e33197/numpy/core/numeric.py#L2280-L2316